app/services/payment_providers/webhooks/stripe/setup_intent_succeeded_service.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module PaymentProviders
+  module Webhooks
+    module Stripe
+      class SetupIntentSucceededService < BaseService
+        include Customers::PaymentProviderFinder
+
+        def call
+          return result if stripe_customer_id.nil?
+          return handle_missing_customer unless stripe_customer
+
+          update_stripe_customer_default_payment_method
+          result.payment_method_id = payment_method_id
+
+          PaymentProviderCustomers::Stripe::UpdatePaymentMethodService.call(
+            stripe_customer:,
+            payment_method_id: payment_method_id
+          ).raise_if_error!
+
+          result.stripe_customer = stripe_customer
+          result
+        rescue ::Stripe::PermissionError => e
+          result.service_failure!(code: 'stripe_error', message: e.message)
+        end
+
+        private
+
+        def stripe_customer
+          @stripe_customer ||= PaymentProviderCustomers::StripeCustomer
```

Comment:
this feels too complicated of a query to have to perform here, should we move this to the model instead? (e.g. someone will forget to join to the organization_id at some point)

Reply:
And I just noticed it's duplicated in some places as well, I think that's a clear sign it should be a scope or finder method 😄